SimpleAggregateFunction initial documentation (#10002)
ClickHouse/ClickHouse: docs/en/data_types/nested_data_structures/simpleaggregatefunction.md

# SimpleAggregateFunction(name, types\_of\_arguments…) {#data-type-simpleaggregatefunction}

> Unlike [`AggregateFunction`](../aggregatefunction.md), which store not the value of the aggregate function but it's state:

Reviewer: ```suggestion
Unlike [`AggregateFunction`](../aggregatefunction.md), which stores not the value of the aggregate function but it's state:
```

> Unlike [`AggregateFunction`](../aggregatefunction.md), which stores not the value of the aggregate function but it's state:
> 
> - `SimpleAggregateFunction` is stateless, so it supports only simple stateless aggregate functions, including:

Reviewer 2: ```suggestion
- `SimpleAggregateFunction` data type stores current value of aggregation and does not store aggregation state as [`AggregateFunction`](../aggregatefunction.md).
.
It supports simple stateless aggregate functions, including:
```

Author: Applied, thanks!

    - [`any`](../../query_language/agg_functions/reference.md#agg_function-any)
    - [`anyLast`](../../query_language/agg_functions/reference.md#anylastx)
    - [`min`](../../query_language/agg_functions/reference.md#agg_function-min)
    - [`max`](../../query_language/agg_functions/reference.md#agg_function-max)
    - [`sum`](../../query_language/agg_functions/reference.md#agg_function-sum)
    - [`groupBitAnd`](../../query_language/agg_functions/reference.md#groupbitand)
    - [`groupBitOr`](../../query_language/agg_functions/reference.md#groupbitor)
    - [`groupBitXor`](../../query_language/agg_functions/reference.md#groupbitxor)

- Type of the `SimpleAggregateFunction(func, Type)` is `Type` itself, so you do not need to apply functions with `-Merge`/`-State` suffixes.

> - Hence has better performance.

Reviewer 2: ```suggestion
- SimpleAggregateFunction has better performance than AggregateFunction with same aggregation function.
```

Author: Applied (and added backquotes), thanks!

# SimpleAggregateFunction(name, types\_of\_arguments…) {#data-type-simpleaggregatefunction}

> Unlike `AggregateFunction`:

Reviewer 3: link to article about AggregateFunction + at least brief introduction about what it does for people who don't want to follow it

> - `SimpleAggregateFunction` is stateless, so it supports only simple stateless functions and for now, only the following are supported:

Reviewer 3: ```suggestion
- `SimpleAggregateFunction` is stateless, so it supports only simple stateless aggregate functions, including:
```

>     - `any`

Reviewer 3: links to documentation on each of them

    - `anyLast`
    - `min`
    - `max`
    - `sum`
    - `sumMap`
    - `groupBitAnd`
    - `groupBitOr`
    - `groupBitXor`

- Type of the `SimpleAggregateFunction(func, Type)` is `Type` itself, so you do not need to apply functions with `-Merge`/`-State` suffixes.
- Hence has better performance.

**Parameters**

-   Name of the aggregate function.

> -   Types of the aggregate function arguments.

Reviewer 3: needs an example with multiple arguments

Author: None of functions supports multiple arguments (sumMap had been dropped, since it is not supported yet)
